### DISCUSSION TITLE
Help in serializing AST to JavaScript

### DISCUSSION THREAD

**Discussion by peter17** [syntax-tree/unist#127] [Q&A]
Hi!

I have a JSON file which is a estree AST. I need to serialize it to JavaScript.

I am trying to use https://github.com/syntax-tree/estree-util-to-js for this, but I am not a JavaScript developer and I don't understand how to have the example code run... (copied below from the README):

```
import fs from 'node:fs/promises'
import {parse} from 'acorn'
import {toJs} from 'estree-util-to-js'

const file = String(await fs.readFile('index.js'))

const tree = parse(file, {ecmaVersion: 2022, sourceType: 'module', locations: true})

// @ts-expect-error: acorn is funky but it works fine.
console.log(toJs(tree))
```

I saved that code to test.js and I installed estree-util-to-js using NPM but when I run `node test.js` I get `SyntaxError: Cannot use import statement outside a module`

I copied the code to test.html and replaced a line with `import {toJs} from 'https://esm.sh/estree-util-to-js@2?bundle'` but I can't do the same for `node:fs/promises`...

Any help, please? Thanks a lot in advance. Best regards

**Comment by ChristianMurphy** [comment]
Welcome @peter17!
Sorry you ran into a spot of confusion.

> SyntaxError: Cannot use import statement outside a module

This indicates you have the file marked as CJS.
You can mark it as ESM by either adding `"type": "module"` in your package.json (recommended) or changing the file extension to `.mjs`
you can find more information on enabling ESM here https://nodejs.org/api/esm.html#enabling

**Comment by peter17** [comment]
Hi @ChristianMurphy 

Thank you very much! It is working fine, now, with `node test.js`!

Best regards
